integrations/issue_test.go

```diff
@@ -184,3 +187,34 @@ func TestIssueCommentClose(t *testing.T) {
 	val := htmlDoc.doc.Find(".comment-list .comments .comment .render-content p").First().Text()
 	assert.Equal(t, "Description", val)
 }
+
+func TestNewIssueConcurrent(t *testing.T) {
+
+	prepareTestEnv(t)
+	session := loginUser(t, "user2")
+
+	f := func(ctx context.Context, cancel context.CancelFunc, wg *sync.WaitGroup) {
+		defer wg.Done()
+
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+		testNewIssue(t, session, "user2", "repo1", fmt.Sprintf("Title"), "Description")
+		if t.Failed() {
```

Comment:
I think a great deal of times this function will fail for reasons unrelated to a duplicate key (which is what this test is for) and give false negatives. It's important to check the error and verify it is actually a duplicate key; other errors should be ignored. Gitea will be built in all kinds of environments, e.g. small Raspberry Pi boards.
Also, this might extend the test times beyond the alloted values in the drone (something to be dealt with).

Reply:
That's true. But I am afraid that there are some other undiscovered bugs, so an integration test might be helpful to an extent if it can reveal them.